components/BottomNavigation/src/private/MDCBottomNavigationSystemDialogView.m

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCBottomNavigationSystemDialogView.h"
+
+#import <CoreGraphics/CoreGraphics.h>
+
+static const CGFloat kCornerRadius = 10;
+static const CGFloat kMargins = 10;
+
+static UIVisualEffectView *MDCInitializeCompatibleBlurView() {
+  if (@available(iOS 10, *)) {
+    UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleProminent];
+    UIVisualEffectView *blurView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    return blurView;
+  }
+
+  UIBlurEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
+  return [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+}
+
+@interface MDCBottomNavigationSystemDialogView ()
+
+/**
+ * The background blur view. This property is nil if on instantiation reduced transparency was
+ * enabled.
+ */
+@property(nonatomic, nullable) UIVisualEffectView *blurView;
+
+@end
+
+@implementation MDCBottomNavigationSystemDialogView
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonInitMDCBottomNavigationSystemDialogView];
+  }
+
+  return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonInitMDCBottomNavigationSystemDialogView];
+  }
+
+  return self;
+}
+
+- (void)commonInitMDCBottomNavigationSystemDialogView {
+  self.layer.cornerRadius = kCornerRadius;
+  self.layer.masksToBounds = YES;
+
+  CGFloat margin = kCornerRadius + kMargins;
+  UIEdgeInsets layoutMargins = UIEdgeInsetsMake(margin, margin, margin, margin);
+  if (UIAccessibilityIsReduceTransparencyEnabled()) {
+    self.backgroundColor = [UIColor colorWithWhite:(CGFloat)0.8 alpha:(CGFloat)1];
+    self.layoutMargins = layoutMargins;
+  } else {
+    _blurView = MDCInitializeCompatibleBlurView();
+    _blurView.contentView.layoutMargins = layoutMargins;
```

Comment:
Will the blur view not inherit the layoutMargins from its view hierarchy?  It would be nice if we could just set `self.layoutMargins`. If not, let's document why that seems to be the case.

Reply:
I believe layout margins only propagate to a view's subviews if the subview has `preservesSuperviewLayoutMargins` set to `YES`.  I am hesitant to use this approach, because a change in the view hierarchy of `UIVisualEffectView`  may break the margins functionality.  Thoughts?

Reply:
I think that's a fair reason. You may want to set `translatesAutoresizingMaskIntoConstraints = NO;` to make it clear you're expecting to do manual layout.